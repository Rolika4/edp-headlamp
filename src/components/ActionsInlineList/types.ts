import { KubeObjectAction } from '../../types/actions';

export interface ActionsInlineListProps {
  actions: KubeObjectAction[];
}
