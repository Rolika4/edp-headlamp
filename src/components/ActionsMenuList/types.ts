import { KubeObjectAction } from '../../types/actions';

export interface ActionsMenuListProps {
  actions: KubeObjectAction[];
  handleCloseActionsMenu?(): void;
  anchorEl: HTMLElement | null;
}
