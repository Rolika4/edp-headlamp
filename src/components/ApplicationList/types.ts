import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';

export interface ApplicationListProps {
    applications: EDPCodebaseKubeObjectInterface[];
    onEdit?(): void;
    onDelete?(): void;
}
