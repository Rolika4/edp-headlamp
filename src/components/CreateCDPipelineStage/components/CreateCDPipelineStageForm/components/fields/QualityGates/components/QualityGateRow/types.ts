import { Autotest, QualityGate } from '../../types';

export interface QualityGateRowProps {
    autotests: Autotest[];
    currentQualityGateData: QualityGate;
    setQualityGates(prev: any): QualityGate[];
    setNewQualityGates(prev: any): QualityGate[];
}
