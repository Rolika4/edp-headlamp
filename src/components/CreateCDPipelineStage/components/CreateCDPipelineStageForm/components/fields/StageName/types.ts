import { CreationFormFieldInterface } from '../../../../../../../types/forms';

export interface StageNameProps extends CreationFormFieldInterface {}
