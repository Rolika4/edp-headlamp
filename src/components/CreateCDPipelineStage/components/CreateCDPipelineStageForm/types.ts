import { EDPCDPipelineKubeObjectInterface } from '../../../../k8s/EDPCDPipeline/types';
import { EDPCDPipelineStageKubeObjectInterface } from '../../../../k8s/EDPCDPipelineStage/types';
import { DeepPartial } from '../../../../types/global';

export interface CreateCDPipelineStageFormProps {
    availableCITools: string[];
    CDPipelineData: DeepPartial<EDPCDPipelineKubeObjectInterface>;
    otherStages: DeepPartial<EDPCDPipelineStageKubeObjectInterface>[];
    editorOpen: boolean;
    setEditorOpen(boolean): void;
    setDialogOpen(boolean): void;
    handleApply: (stage: DeepPartial<EDPCDPipelineStageKubeObjectInterface>) => void;
    isApplying: boolean;
}
