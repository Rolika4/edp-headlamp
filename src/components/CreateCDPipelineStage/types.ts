import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';
import { EDPCDPipelineStageKubeObjectInterface } from '../../k8s/EDPCDPipelineStage/types';
import { DeepPartial } from '../../types/global';

export interface CreateCDPipelineStageProps {
    availableCITools: string[];
    CDPipelineData: DeepPartial<EDPCDPipelineKubeObjectInterface>;
    otherStages: DeepPartial<EDPCDPipelineStageKubeObjectInterface>[];
    open: boolean;
    onClose(): void;
    setOpen(boolean): void;
    handleApply: (stage: DeepPartial<EDPCDPipelineStageKubeObjectInterface>) => void;
    isApplying: boolean;
}
