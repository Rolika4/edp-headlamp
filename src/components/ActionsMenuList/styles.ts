import makeStyles from '@mui/styles/makeStyles';

export const useStyles = makeStyles((theme) => ({
  actions: {
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'center',

    '& .MuiTooltip-popper': {
      pointerEvents: 'auto',
    },
  },
  actionList: {
    backgroundColor: theme.palette.background.paper,
    borderRadius: theme.typography.pxToRem(5),
    overflow: 'hidden',
    padding: `${theme.typography.pxToRem(8)} 0`,
  },
  popper: {
    zIndex: 5,
    cursor: 'default',
    padding: 0,
    backgroundColor: theme.palette.background.paper,
    color: theme.palette.text.primary,
    boxShadow: '0px 5px 5px -3px #00000033, 0px 8px 10px 1px #00000024, 0px 3px 14px 2px #0000001F',
    borderRadius: theme.typography.pxToRem(5),
  },
  childrenWrapper: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
}));
