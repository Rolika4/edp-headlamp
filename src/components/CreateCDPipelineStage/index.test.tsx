/**
 * @jest-environment jsdom
 */

import { describe, jest } from '@jest/globals';
import { createTheme } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/styles';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen, waitFor } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { Provider } from 'react-redux';
import { namespacesMock } from '../../hooks/useNamespaces/mocks/namespaces.mock';
import { pluginLib } from '../../plugin.globals';
import { CreateCDPipelineStage } from './index';
import { CreateCDPipelineStageProps } from './types';

const { ApiProxy } = pluginLib;

describe('CreateCDPipelineStage', () => {
    it('should render correctly', async () => {
        const props: CreateCDPipelineStageProps = {
            availableCITools: ['jenkins', 'tekton'],
            CDPipelineData: {
                apiVersion: 'apiVersion',
                kind: 'CDPipeline',
                metadata: {
                    name: 'test-app-name',
                    namespace: 'test-namespace',
                },
            },
            otherStages: [
                {
                    apiVersion: 'apiVersion',
                    kind: 'Stage',
                    spec: {
                        name: 'sit',
                    },
                    metadata: {
                        name: 'test-pipe-sit',
                        namespace: 'test-namespace',
                    },
                },
            ],
            open: true,
            setOpen: () => {},
            onClose: () => {},
            handleApply: () => {},
            isApplying: false,
        };
        const store = configureStore({
            reducer: () => ({}),
        });

        jest.spyOn(ApiProxy, 'request').mockImplementation(url => {
            if (url.includes('/api/v1/namespaces')) {
                return Promise.resolve({
                    items: [
                        { kind: 'Namespace', metadata: { name: 'namespace-1' } },
                        { kind: 'Namespace', metadata: { name: 'namespace-2' } },
                    ],
                });
            }
        });

        const theme: DefaultTheme = createTheme();

        jest.spyOn(ApiProxy, 'request').mockImplementation(url => {
            if (url.includes('/api/v1/namespaces')) {
                return Promise.resolve(namespacesMock);
            }
        });

        render(
            <Provider store={store}>
                <SnackbarProvider>
                    <ThemeProvider theme={theme}>
                        <CreateCDPipelineStage {...props} />
                    </ThemeProvider>
                </SnackbarProvider>
            </Provider>
        );

        await waitFor(() => {
            expect(screen.getByTestId('create-cdpipeline-stage')).toMatchSnapshot();
        });
    });
});
