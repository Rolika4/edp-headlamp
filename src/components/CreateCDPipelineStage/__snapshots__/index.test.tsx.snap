// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCDPipelineStage should render correctly 1`] = `
<div
  class="makeStyles-dialog-2"
  data-testid="create-cdpipeline-stage"
>
  <div
    class="makeStyles-dialogTitle-3"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      Create stage for "test-app-name"
    </h5>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall"
        >
          <span />
        </span>
        Edit YAML
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialogContent-root makeStyles-dialogContent-4"
  >
    <div
      class="makeStyles-form-5"
    >
      <form>
        <div
          class="makeStyles-formInner-6"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-end"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <span
                        class="makeStyles-labelWrap-8"
                      >
                        <span
                          class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                        >
                          Stage name
                        </span>
                        <span />
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex; align-items: flex-end;"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiFormControl-root MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input MuiInput-input"
                              name="name"
                              placeholder="Enter stage name"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <span
                        class="makeStyles-labelWrap-8"
                      >
                        <span
                          class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                        >
                          Description
                        </span>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex; align-items: flex-end;"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiFormControl-root MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input MuiInput-input"
                              name="description"
                              placeholder="Enter stage description"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <span
                        class="makeStyles-labelWrap-8"
                      >
                        <span
                          class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                        >
                          Trigger type
                        </span>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="mui-component-select-triggerType"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                            id="mui-component-select-triggerType"
                            role="button"
                            tabindex="0"
                          >
                            Manual
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            name="triggerType"
                            tabindex="-1"
                            value="Manual"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <span
                        class="makeStyles-labelWrap-8"
                      >
                        <span
                          class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                        >
                          Job provisioner
                        </span>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="mui-component-select-jobProvisioning"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                            id="mui-component-select-jobProvisioning"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            name="jobProvisioning"
                            tabindex="-1"
                            value="default"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-8"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                            >
                              Groovy pipeline library
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-sourceLibraryName"
                                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-sourceLibraryName"
                                role="button"
                                tabindex="0"
                              >
                                Default
                              </div>
                              <input
                                aria-hidden="true"
                                class="MuiSelect-nativeInput"
                                name="sourceLibraryName"
                                tabindex="-1"
                                value="default"
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-8"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                            >
                              Branch
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-selectWithDefaultValue-10 Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                            >
                              <div
                                aria-disabled="true"
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-sourceLibraryBranch"
                                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled Mui-disabled"
                                id="mui-component-select-sourceLibraryBranch"
                                role="button"
                              >
                                Default
                              </div>
                              <input
                                aria-hidden="true"
                                class="MuiSelect-nativeInput"
                                name="sourceLibraryBranch"
                                tabindex="-1"
                                value=""
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6"
                  >
                    Quality gates
                  </h6>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-end"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-container"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                              >
                                <div
                                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                                >
                                  <div
                                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                                  >
                                    <div
                                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                      style="display: flex;"
                                    >
                                      <div
                                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                                      >
                                        <div
                                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                        >
                                          <span
                                            class="makeStyles-labelWrap-8"
                                          >
                                            <span
                                              class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                                            >
                                              Quality gate type
                                            </span>
                                          </span>
                                        </div>
                                        <div
                                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                        >
                                          <div
                                            class="MuiFormControl-root MuiFormControl-fullWidth"
                                          >
                                            <div
                                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                                            >
                                              <div
                                                aria-haspopup="listbox"
                                                aria-labelledby="mui-component-select-quality-gate-type-0"
                                                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                                id="mui-component-select-quality-gate-type-0"
                                                role="button"
                                                tabindex="0"
                                              >
                                                Manual
                                              </div>
                                              <input
                                                aria-hidden="true"
                                                class="MuiSelect-nativeInput"
                                                name="quality-gate-type-0"
                                                tabindex="-1"
                                                value="manual"
                                              />
                                              <svg
                                                aria-hidden="true"
                                                class="MuiSvgIcon-root MuiSelect-icon"
                                                focusable="false"
                                                viewBox="0 0 24 24"
                                              >
                                                <path
                                                  d="M7 10l5 5 5-5z"
                                                />
                                              </svg>
                                            </div>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                                <div
                                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                                >
                                  <div
                                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                                  >
                                    <div
                                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                      style="display: flex;"
                                    >
                                      <div
                                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                                      >
                                        <div
                                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                        >
                                          <span
                                            class="makeStyles-labelWrap-8"
                                          >
                                            <span
                                              class="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
                                            >
                                              Step name
                                            </span>
                                          </span>
                                        </div>
                                        <div
                                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                          style="display: flex; align-items: flex-end;"
                                        >
                                          <div
                                            class="MuiFormControl-root MuiFormControl-fullWidth"
                                          >
                                            <div
                                              class="MuiFormControl-root MuiTextField-root"
                                            >
                                              <div
                                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                                              >
                                                <input
                                                  aria-invalid="false"
                                                  class="MuiInputBase-input MuiInput-input"
                                                  name="quality-gate-step-name-0"
                                                  placeholder="Enter step name"
                                                  type="text"
                                                  value=""
                                                />
                                              </div>
                                            </div>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    style="min-width: 0;"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      add
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation2"
                    role="alert"
                  >
                    <div
                      class="MuiAlert-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiAlert-message"
                    >
                      Add at least one quality gate
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-actions-7"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              undo changes
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            style="margin-left: auto;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              apply
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
